esports_data_from_rimble/rimble/rimble.go

```diff
@@ -0,0 +1,201 @@
+package rimble
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/samber/lo"
+)
+
+type PlayerResult struct {
+	Kills      int     `json:"kills"`
+	GameNumber int     `json:"game_number"`
+	KAST       float64 `json:"KAST"`
+	Deaths     int     `json:"deaths"`
+	ADR        float64 `json:"ADR"`
+}
+
+type Player struct {
+	Name     string         `json:"name"`
+	ID       string         `json:"id"`
+	Results  []PlayerResult `json:"results"`
+	Username string         `json:"username"`
+}
+
+type TeamResult struct {
+	GameNumber int `json:"game_number"`
+	TotalKills int `json:"totalKills"`
+	RoundsWon  int `json:"rounds_won"`
+}
+
+type Team struct {
+	GamesWon    int          `json:"games_won"`
+	WinResult   int          `json:"win_result"`
+	Players     []Player     `json:"players"`
+	Name        string       `json:"name"`
+	ID          string       `json:"id"`
+	Designation int          `json:"designation"`
+	Results     []TeamResult `json:"results"`
+}
+
+type Game struct {
+	GameNumber int    `json:"game_number"`
+	MapName    string `json:"map_name"`
+}
+
+type Metadata struct {
+	Games []Game `json:"games"`
+}
+
+type MatchData struct {
+	Metadata    Metadata `json:"metadata"`
+	Team1Name   string   `json:"team_1_name"`
+	Team2Name   string   `json:"team_2_name"`
+	Teams       []Team   `json:"teams"`
+	MatchID     string   `json:"matchid"`
+	MatchStatus string   `json:"match_status"`
+}
+
+func MakeMatchDataFromMatchesJSON(matchesJSON []byte) (MatchData, error) {
+	var matches []MatchData
+	err := json.Unmarshal(matchesJSON, &matches)
+	if err != nil {
+		return MatchData{}, fmt.Errorf("error unmarshaling JSON: %w", err)
+	}
+
+	switch len(matches) {
+	case 0:
+		err = fmt.Errorf("found no matches")
+		return MatchData{}, err
+	case 1:
+		break // only one match found, proceed to return it
+	default:
+		err = fmt.Errorf("found multiple matches")
+		return MatchData{}, err
+	}
+
+	return matches[0], nil
+}
+
+func (match MatchData) Winner() (Team, error) {
+	winningTeams := lo.Filter(match.Teams, func(team Team, _ int) bool {
```

Comment:
I decided to use the `lo` package to make the parsing easier. I was curious if it would work work in our functions and even with it imported the function size is still pretty small, about 2.4MB.